src/components/device/boxfish.ts

```diff
@@ -152,6 +153,26 @@ export default class Boxfish extends UvcBaseDevice implements IDeviceManager {
     return createBoxfishUpgrader(this, this.discoveryEmitter, this.logger);
   }
 
+  async createAndRunFsblUpgrade(opts: UpgradeOpts, deviceManager: IDeviceManager) {
+    const upgrader = new BoxfishUpgrader(deviceManager, this.discoveryEmitter, this.logger);
+    const mvusbFile = opts.file;
+    const timeoutMs = opts.bootTimeout * 1000;
+
+    return new Promise((resolve, reject) => {
+      const bootTimeout = setTimeout(() => {
+        clearTimeout(bootTimeout);
+        reject('Fslb upgrade timed out');
+      }, timeoutMs);
+
+      upgrader.flashFsbl(mvusbFile).then(
```

Comment:
Use arrow style function declaration
```suggestion
      upgrader.flashFsbl(mvusbFile).then(() => { clearTimeout(bootTimeout);resolve();})
```